modules/caddyhttp/encode/caddyfile.go

```diff
@@ -64,24 +74,147 @@ func (enc *Encode) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 
 		for d.NextBlock(0) {
 			name := d.Val()
-			modID := "http.encoders." + name
-			unm, err := caddyfile.UnmarshalModule(d, modID)
-			if err != nil {
-				return err
-			}
-			encoding, ok := unm.(Encoding)
-			if !ok {
-				return fmt.Errorf("module %s is not an HTTP encoding; is %T", modID, unm)
+
+			// if the subdirective has an "@" prefix then we
+			// parse it as a response matcher for use with "match"
+			if strings.HasPrefix(name, matcherPrefix) {
+				err := enc.parseNamedResponseMatcher(d.NewFromNextSegment(), responseMatchers)
+				if err != nil {
+					return err
+				}
+				continue
 			}
-			if enc.EncodingsRaw == nil {
-				enc.EncodingsRaw = make(caddy.ModuleMap)
+
+			switch name {
+			case "minimum_length":
+				if !d.NextArg() {
+					return d.ArgErr()
+				}
+				minLength, err := strconv.Atoi(d.Val())
+				if err != nil {
+					return err
+				}
+				enc.MinLength = minLength
+			case "prefer":
+				var encs []string
+				for d.NextArg() {
+					encs = append(encs, d.Val())
+				}
+				if len(encs) == 0 {
+					return d.ArgErr()
+				}
+				enc.Prefer = encs
+			case "match":
+				var matcher *caddyhttp.ResponseMatcher
+				args := d.RemainingArgs()
+				// the first arg should be a matcher (optional)
+				// the second arg should be a status code (optional)
+				// any more than that isn't currently supported
+				if len(args) > 1 {
+					return d.Errf("too many arguments for 'if_matches': %s", args)
+				}
+
+				// the first arg should always be a matcher.
+				// it doesn't really make sense to support status code without a matcher.
+				if len(args) > 0 {
+					if !strings.HasPrefix(args[0], matcherPrefix) {
+						return d.Errf("must use a named response matcher, starting with '@'")
+					}
+
+					foundMatcher, ok := responseMatchers[args[0]]
+					if !ok {
+						return d.Errf("no named response matcher defined with name '%s'", args[0][1:])
+					}
+					matcher = &foundMatcher
+				}
+				enc.Matcher = matcher
+			default:
+				modID := "http.precompressed." + name
+				unm, err := caddyfile.UnmarshalModule(d, modID)
+				if err != nil {
+					return err
+				}
+				encoding, ok := unm.(Encoding)
+				if !ok {
+					return d.Errf("module %s is not an HTTP encoding; is %T", modID, unm)
+				}
+				if enc.EncodingsRaw == nil {
+					enc.EncodingsRaw = make(caddy.ModuleMap)
+				}
+				enc.EncodingsRaw[name] = caddyconfig.JSON(encoding, nil)
 			}
-			enc.EncodingsRaw[name] = caddyconfig.JSON(encoding, nil)
 		}
 	}
 
 	return nil
 }
 
+// Parse the tokens of a named response matcher.
+//
+//     @name {
+//         header <field> [<value>]
+//         status <code...>
+//     }
+//
+// Or, single line syntax:
+//
+//     @name [header <field> [<value>]] | [status <code...>]
+//
+func (enc *Encode) parseNamedResponseMatcher(d *caddyfile.Dispenser, matchers map[string]caddyhttp.ResponseMatcher) error {
```

Comment:
We'll probably want to move this somewhere common, but since it's internal I suppose it can wait until both this PR and the `handle_response` one is merged.

Reply:
Sounds good to me. I just shamelessly copied it 😇